src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -54,6 +55,27 @@ if (__DEV__) {
       callback,
     );
   };
+
+  // This is so gross but it's at least non-critical and can be removed if
+  // it causes problems. This is meant to give a nicer error message for
+  // ReactDOM15.unstable_renderSubtreeIntoContainer(reactDOM16Component,
+  // ...)) which otherwise throws a "_processChildContext is not a function"
+  // exception.
+  Object.defineProperty(fakeInternalInstance, '_processChildContext', {
+    enumerable: false,
+    value: function() {
+      invariant(
+        false,
+        '_processChildContext is not available in React 16+. This likely ' +
+          'means you have multiple copies of React and are attempting to nest ' +
+          'a React 15 tree inside a React 16 tree using ' +
+          "unstable_renderSubtreeIntoContainer, which isn't supported. Try " +
+          'to make sure you have only one copy of React (and ideally, switch ' +
+          'to ReactDOM.unstable_createPortal).',
```

Comment:
nit - we could condense this and then link to docs, which could provide more info.

I think we could always improve the error message in a follow-up commit, if this comes up often for folks.